x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -211,6 +214,49 @@ export const signalRulesAlertType = ({
           await ruleStatusService.error(errorString);
           wroteStatus = true;
         }
+        const timestampFieldCaps = await services.scopedClusterClient.fieldCaps({
+          index,
+          fields: timestampFields,
+          allow_no_indices: false,
+          include_unmapped: true,
+        });
+
+        if (
+          !wroteStatus &&
+          params.timestampOverride != null &&
+          (isEmpty(timestampFieldCaps?.body?.fields) ||
+            timestampFieldCaps?.body?.fields[params.timestampOverride] == null ||
+            timestampFieldCaps?.body?.fields[params.timestampOverride]?.unmapped?.indices != null)
+        ) {
+          // if there is a timestamp override and the unmapped array for the timestamp override key is not empty,
+          // partial failure
+          const errorString = `The following indices are missing the timestamp override field "${
+            params.timestampOverride
+          }": ${JSON.stringify(
+            isEmpty(timestampFieldCaps?.body?.fields)
+              ? timestampFieldCaps.body.indices
+              : timestampFieldCaps?.body?.fields[params.timestampOverride]?.unmapped?.indices
+          )}`;
+          logger.debug(buildRuleMessage(errorString));
+          await ruleStatusService.partialFailure(errorString);
+          wroteStatus = true;
+        } else if (
+          !wroteStatus &&
+          params.timestampOverride == null &&
+          (isEmpty(timestampFieldCaps?.body?.fields) ||
+            timestampFieldCaps?.body?.fields['@timestamp'].unmapped?.indices != null)
+        ) {
+          // if there is no timestamp override and the unmapped array is not empty,
+          // partial failure
+          const errorString = `The following indices are missing the timestamp field "@timestamp": ${JSON.stringify(
+            isEmpty(timestampFieldCaps?.body?.fields)
+              ? timestampFieldCaps.body.indices
+              : timestampFieldCaps?.body?.fields['@timestamp']?.unmapped?.indices
+          )}`;
+          logger.debug(buildRuleMessage(errorString));
+          await ruleStatusService.partialFailure(errorString);
+          wroteStatus = true;
```

Comment:
This variable and the `hasError` above it aren't really two separate things from looking at this logic. It looks like you could actually remove the `wroteStatus` and just use the `hasError` above, no?

Doesn't have to be this PR but looking through this, I don't think you need the extra variable. I could be missing something though.

Reply:
The `hasError` looks like it was introduced to handle gap detection errors and errors with starting the ML jobs. It appears we need a better design around when / how we are writing error statuses and handling that logic within the executor. I'm not sure we can do a direct replace with `wroteStatus` and `hasError` at this time without doing an overhaul of how we are managing different error states within the executor. 

For instance if we were to use `hasError` and drop `wroteStatus` we could have an error with an ML rule where it couldn't read an index **and** the ml job failed to start. Which failure takes precedence as the failure to be written to the current status? I think we need a way to set precedence when writing errors. At this time it was easier to add a secondary variable than working on a new system for writing rule statuses. But for sure this needs to be revisited again.